src/seedu/addressbook/storage/jaxb/AdaptedPerson.java

```diff
@@ -93,14 +93,19 @@ public boolean isAnyRequiredFieldMissing() {
      * @throws IllegalValueException if there were any data constraints violated in the adapted person
      */
     public Person toModelType() throws IllegalValueException {
+        String[] fullAddress = this.address.value.split(",");
+        String blockName = fullAddress[0];
+        String streetName = fullAddress[1];
+        String unitName = fullAddress[2];
+        String postalName = fullAddress[3];
         final List<Tag> personTags = new ArrayList<>();
         for (AdaptedTag tag : tagged) {
             personTags.add(tag.toModelType());
         }
         final Name name = new Name(this.name);
         final Phone phone = new Phone(this.phone.value, this.phone.isPrivate);
         final Email email = new Email(this.email.value, this.email.isPrivate);
-        final Address address = new Address(this.address.value, this.address.isPrivate);
+        final Address address = new Address(blockName,streetName,unitName,postalName, this.address.isPrivate);
```

Comment:
see the problem of splitting the address in `AddCommand`? you have to make the change in this class as well.